apps/api/views.py

```diff
@@ -548,6 +550,16 @@ def trigger_bot_message(request):
 
     if platform == ChannelPlatform.COMMCARE_CONNECT and not participant_data.has_consented():
         return JsonResponse({"detail": "User has not given consent"}, status=status.HTTP_400_BAD_REQUEST)
+    if data.get("participant_data"):
+        if participant_data:
+            merged_data = {**participant_data.data, **data["participant_data"]}
+            participant_data.data = merged_data
+            participant_data.save(update_fields=["data"])
+        else:
+            participant = Participant.objects.filter(identifier=identifier, platform=platform).first()
```

Comment:
This needs to filter by team. Also rather use `.get` which will fail loudly if the participant doesn't exist.
```suggestion
            participant = Participant.objects.get(team=request.team, identifier=identifier, platform=platform)
```